src/robot.coffee

```diff
@@ -68,6 +68,20 @@ class Robot
       @emit 'error', err
     process.on 'uncaughtException', @onUncaughtException
 
+  # Public: Adds a custom Listener with the provided matcher, options, and
+  # callback
+  #
+  # matcher  - A Function that determines whether to call the callback.
+  #            Expected to return a boolean.
```

Comment:
It's not _just_ a boolean. It matches if it's truthy, and that truthy value is included as `response.match`